packages/rich-text/src/component/index.js

```diff
@@ -947,29 +956,22 @@ class RichText extends Component {
 
 	Editable( props ) {
```

Comment:
While I understand a goal of this pull request was to eliminate an "Editable" component, that's effectively what I see this function as being, and perhaps we should still have it externalized from the parent?

One potential advantage could be in enabling you to use hooks.

Reply:
The goal wasn't to eliminate `Editable`, but to have React manage the element. A side effect of that is that `Editable` is nothing more than just the element. What logic would move to the function in the form of hooks?

Reply:
Well, I had a few initial thoughts around how we could `useMemo` for caching the combined style object (turned out maybe not viable, or at least the stylesheet idea could work better) or `useCallback` for the assigned value of `onKeyDown` (turned out if the component was made separate, this logic would have to occur in the parent `RichText` anyways).

tl;dr I wouldn't worry too much about it now, but maybe in the future there could be some additional requirements where hooks would prove to be useful.

Reply:
Yeah, I also agree that this component could use some more splitting in general. It's definitely on my mind.